### ISSUE TITLE
Add @ExpectMaxInsert

### ISSUE THREAD

**Issue by jeanbisutti** [quick-perf/quickperf#67]
**Description**

The aim of this annotation is to check the maximum number of INSERT statements.

For example, a test method having `@ExpectMaxInsert(2)` annotation will fail if more than two INSERT statements are executed.

**Implementation steps**

1) Create the `@ExpectMaxInsert` annotation [here](https://github.com/quick-perf/quickperf/tree/master/sql/sql-annotations/src/main/java/org/quickperf/sql/annotation)

    Example: [@ExpectInsert](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/annotation/ExpectInsert.java)

2) Create a test class in [this package](https://github.com/quick-perf/quickperf/tree/master/sql/sql-integration-test/src/test/java) and add automatic tests

    Example: [ExpectInsertTest.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-integration-test/src/test/java/ExpectInsertTest.java)

3) In this [package](https://github.com/quick-perf/quickperf/tree/master/sql/sql-annotations/src/main/java/org/quickperf/sql/insert) add a performance measure extractor and a performance issue verifier. 

    Examples: 
    * [InsertCountMeasureExtractor.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/insert/InsertCountMeasureExtractor.java)
    * [InsertNumberPerfIssueVerifier.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/insert/InsertNumberPerfIssueVerifier.java)  
     * [MaxOfSelectsPerfIssueVerifier.java](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/select/MaxOfSelectsPerfIssueVerifier.java)

4) Create an AnnotationConfig in [SqlAnnotationsConfig](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/config/library/SqlAnnotationsConfigs.java). This new AnnotationConfig will use the previously created performance measure extractor, performance issue verifier together with a PersistenceSqlRecorder.

    Example:
    ```java
        static final AnnotationConfig MAX_SQL_SELECT = new AnnotationConfig.Builder()
            .perfRecorderClass(PersistenceSqlRecorder.class)
            .perfMeasureExtractor(SelectCountMeasureExtractor.INSTANCE)
            .perfIssueVerifier(MaxOfSelectsPerfIssueVerifier.INSTANCE)
            .build(ExpectMaxSelect.class);
     ```

5) Use the new AnnotationConfig in `loadAnnotationConfigs()` method of [SqlConfigLoader](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/config/library/SqlConfigLoader.java) 

6) Complete the measure extractor and the performance issue verifier to make the tests green. Refactor to clean the code.

7) In [SqlAnnotationBuilder](https://github.com/quick-perf/quickperf/blob/master/sql/sql-annotations/src/main/java/org/quickperf/sql/annotation/SqlAnnotationBuilder.java), add a static method to can build the new annotation. This method can be used to define the annotation with [a global scope](https://github.com/quick-perf/doc/wiki/QuickPerf#annotation-scopes).

If you start to work on this issue, please leave a comment _I start to work on this issue_, so we can assign it to you. Do not hesitate to ask for information about the issue.

**Comment by MrSnix** [comment]
I start to work on this issue

**Comment by MrSnix** [comment]
I'm not sure what i should write inside HibernateSuggestion on N_PLUS_ONE_INSERT...
Should i document a way on how to control this type of behavior for Hibernate, 
just as it was done for N_PLUS_ONE_SELECT ?
Is it supposed to display advice on how to solve this behavior?

**Comment by jeanbisutti** [comment]
Great @MrSnix!
I assign this issue to you.
No suggestion was asked for this annotation. I didn't think about this when I created this issue. A suggestion may be "Perhaps you may think that JDBC batching is enabled. You can check this with  @ExpectJdbcBatching". 
[@ExpectJdbcBatching documentation](https://github.com/quick-perf/doc/wiki/@ExpectJdbcBatching)

**Comment by jeanbisutti** [comment]
Fixed with 25a39ff7b15e10639d86219d166a79aaed2708a5

**Comment by jeanbisutti** [comment]
Many thanks @MrSnix!
